src/js/line.js

```diff
@@ -0,0 +1,491 @@
+            description: ".floec-ca-lineChart-description",
+            svg: ".floec-ca-lineChart-line",
+            yAxis: ".floec-ca-lineChart-y-axis",
+            xAxis: ".floec-ca-lineChart-x-axis",
+            chartLine: ".floec-ca-lineChart-chartLine",
+            chartLinePointGroup: ".floec-ca-lineChart-chartLinePointGroup",
+            chartLinePoint: ".floec-ca-lineChart-chartLinePoint",
+            chartLineArea: ".floec-ca-lineChart-chartLineArea"
+        },
+        events: {
+            onChartCreated: null,  // Fire when the line is created. Ready to register D3 DOM event listeners,
+            onChartRedrawn: null // Fire when the chart is redrawn.
+        },
+        listeners: {
+            "onCreate.create": {
+                funcName: "floe.chartAuthoring.lineChart.chart.create",
+                args: ["{that}"]
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        modelRelay: {
+            source: "{that}.model.dataSet",
+            target: "{that}.model.wrappedDataSet",
+            singleTransform: {
+                type: "fluid.transforms.free",
+                args: ["{that}.model.dataSet"],
+                func: "floe.chartAuthoring.lineChart.chart.wrapSingleDataSet"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "floe.chartAuthoring.lineChart.chart.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // Test if a dataset is multi
+    floe.chartAuthoring.lineChart.chart.isMultiDataSet = function (dataSet) {
+        return dataSet[0].id !== undefined && dataSet[0].data !== undefined;
+    };
+
+    // Wrap a non-multi dataset (a simple array without ID keys) so we can
+    // process it with the same function used for multi datasets, and
+    // create a default id for it for object constancy
+    floe.chartAuthoring.lineChart.chart.wrapSingleDataSet = function (dataSet) {
+        if (!floe.chartAuthoring.lineChart.chart.isMultiDataSet(dataSet)) {
+            dataSet = [{
+                id: "dataSet",
+                data: dataSet
+            }];
+        }
+        return dataSet;
+    };
+
+    floe.chartAuthoring.lineChart.chart.draw = function (that) {
+
+        var shouldAddArea = that.options.lineOptions.addArea,
+            shouldAddPoints = that.options.lineOptions.addPoints;
+
+        that.yScale = floe.chartAuthoring.lineChart.chart.getYScale(that);
+
+        that.xScale = floe.chartAuthoring.lineChart.chart.getXScale(that);
+
+        that.yAxis = floe.chartAuthoring.lineChart.chart.getYAxis(that);
+
+        that.xAxis = floe.chartAuthoring.lineChart.chart.getXAxis(that);
+
+        that.line = floe.chartAuthoring.lineChart.chart.getLineGenerator(that);
+
+        that.area = floe.chartAuthoring.lineChart.chart.getAreaGenerator(that);
```

Comment:
There are some variables that are tracked by being attached to ```that``` but they can potentially be private function variables. For example, ```area``` can be calculated and added in a function when ```shouldAddArea``` is ```true```. The same question for ```that.line```, ```that.xAxis```, ```that.yAxis``` as well as ```that.chartLinePointGroups``` at line 292. There might be other vars that I miss. Is there a reason that you'd like to track all those values.